src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3803 @@
+export presentation
+
+# TODO replace asserts by error messages?
+
+@doc Markdown.doc"""
+  ModuleFP{T}
+
+The abstract supertype of all modules. Here, all modules are finitely presented.
+The type variable `T` refers to the type of the elements of the base ring.
+"""
+abstract type ModuleFP{T} end
+
+@doc Markdown.doc"""
+  CRing
+
+Type union for commutative rings.
```

Comment:
Isn't this what `Ring` and `RingElem` are (in AA these are *commutative*, and non-commutative rings are `NCRing` -- this is something I'd like to change, but for now it, is how it is).


CC @fieker @thofam

Reply:
It seems that here a "commutative ring" is a ring which has something do to with multivariate polynomials. Seems unrelated to the notion of commutative rings or rings in general.

Reply:
For the moment we aim at multivariate polynomial rings, quotients and localizations thereof and completions of the latter. The code should also work over Euclidean rings. The cases beyond polynomial rings still require some modifications in the code. In the end, we need a class of computational commutative rings which allow kernel computation.  Generalization to non-commutative is possible.

Reply:
I don't think we should introduce commutative rings like this. It would be best to just use `Ring` instead of `CRing` everywhere, which as @fingolfin mentions is the type for commutative rings. At the moment we have no clean way to restrict it to "computational rings which allow kernel computation", so `Ring/RingElem` it should be.